lib/features/family/features/reflect/presentation/pages/grateful_screen.dart

```diff
@@ -238,13 +233,17 @@ class _GratefulScreenState extends State<GratefulScreen> {
           },
         ),
       );
-      _give.add(
-        GiveAmountChanged(
-          firstCollectionAmount: result.toDouble(),
-          secondCollectionAmount: 0,
-          thirdCollectionAmount: 0,
+      await _give.createTransaction(
+        transaction: Transaction(
```

Comment:
The only thing I would do differently is let the cubit create the transaction instead of the widget. Now the widget is a bit too smart and every widget that uses it needs to know to base64encode etc.